### PR TITLE
Search: Fix select item when pressing enter and Grafana is served using a sub path

### PR DESCRIPTION
**Pull request by tskarhed**
**What this PR does / why we need it**:
Fixes 404 when you serve Grafana from a sub url and press enter in search.

- Enable `serve_from_sub_path = true` in config.
- Serve from a subpath

### CODE REVIEW

File: public/app/features/search/hooks/useDashboardSearch.ts [hunk 1]
```diff
@@ -31,7 +32,11 @@ export const useDashboardSearch = (query: DashboardQuery, onCloseSearch: () => v
           if (selectedItem.type === DashboardSearchItemType.DashFolder) {
             onToggleSection(selectedItem as DashboardSection);
           } else {
-            getLocationSrv().update({ path: selectedItem.url });
+            getLocationSrv().update({
+              path: selectedItem.url.startsWith(config.appSubUrl)
+                ? selectedItem.url.split(config.appSubUrl + '/')[1]
+                : selectedItem.url,
+            });
```
Comment (Clarity-89) [inline public/app/features/search/hooks/useDashboardSearch.ts:39]:
```suggestion
            const { appSubUrl } = config;
            const { url } = selectedItem;
            getLocationSrv().update({
              path: url.startsWith(appSubUrl) ? url.split(appSubUrl + '/')[1] : url,
            });
```
This looks a bit cleaner imo.

**Review by Clarity-89 — APPROVED** [review]
Thanks for fixing this! Didn't know about `appSubUrl`. Left a small nit to consider.

File: public/app/features/search/hooks/useDashboardSearch.ts [hunk 2]
```diff
@@ -31,7 +32,11 @@ export const useDashboardSearch = (query: DashboardQuery, onCloseSearch: () => v
           if (selectedItem.type === DashboardSearchItemType.DashFolder) {
             onToggleSection(selectedItem as DashboardSection);
           } else {
-            getLocationSrv().update({ path: selectedItem.url });
+            getLocationSrv().update({
+              path: selectedItem.url.startsWith(config.appSubUrl)
+                ? selectedItem.url.split(config.appSubUrl + '/')[1]
+                : selectedItem.url,
```
Comment (dprokop) [inline public/app/features/search/hooks/useDashboardSearch.ts:38]: think you can use `locationUtil.assureBaseUrl` function from grafana/data for this :)

Comment (tskarhed) [reply public/app/features/search/hooks/useDashboardSearch.ts:38]: I remember that you mentioned it before, but I couldn't find it!

**Pull request merged**